test/Pose.Tests/ShimTests.cs

```diff
@@ -59,5 +62,39 @@ public void TestShimReplaceWith()
             Assert.AreSame(shimTests, shim1.Instance);
             Assert.AreEqual(actionInstance, shim1.Replacement);
         }
+
+        [TestMethod]
+        public void TestReplacePropertyGetter()
+        {
+            Shim shim = Shim.Replace(() => Thread.CurrentThread.CurrentCulture);
+
+            Assert.AreEqual(typeof(Thread).GetProperty(nameof(Thread.CurrentCulture), typeof(CultureInfo)).GetMethod, shim.Original);
+            Assert.IsNull(shim.Replacement);
+        }
+
+        /// <summary>
+        /// The C# compiler currently does not allow poperty setters in lambda expressions.
+        /// This helper method works around that by combining a property getter expression
+        /// and a value into a setter expression.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="lvalue"></param>
+        /// <param name="rvalue"></param>
+        /// <returns></returns>
+        private static Expression<Action> Assignment<T>(Expression<Func<T>> lvalue, T rvalue)
+        {
+            var value = Expression.Constant(rvalue, typeof(T));
+            var assign = Expression.Assign(lvalue.Body, value);
+            return Expression.Lambda<Action>(assign);
+        }
+
+        [TestMethod]
+        public void TestReplacePropertySetter()
+        {
+            Shim shim = Shim.Replace(Assignment(() => Is.A<Thread>().CurrentCulture, Is.A<CultureInfo>()));
+
+            Assert.AreEqual(typeof(Thread).GetProperty(nameof(Thread.CurrentCulture), typeof(CultureInfo)).SetMethod, shim.Original);
+            Assert.IsNull(shim.Replacement);
+        }
```

Comment:
Please add the `TestReplacePropertySetterAction` you mentioned in the comments

Reply:
Ok, I added it to ShimTests

Reply:
Sweet. I'll marge this after the next release goes out. Hopefully, by then I'd have figured out the setter syntax. Any ideas you might have are also welcome